### DISCUSSION TITLE
Running this on a microcontroller or other tiny machine?

### DISCUSSION THREAD

**Discussion by ghost** [ggml-org/whisper.cpp#783] [General]
I would like to hack voice recognition into toys.

To do so, I would need to run this on the smallest possible device, and run it with battery.

I presume at this stage that Raspberry Pi Zero is the smallest machine that this can run on?

Has anyone managed to get it to run on anything smaller?  My guess is that there is no microcontroller with enough memory to run Whisper/C++ is that correct?

Has anyone actually run it successfully on a Raspberry Pi Zero or Zero 2?

**Comment by panamantis** [comment]
I doubt Zero 1 would work -- you'll need the Zero 2 4-cores (raspberry pi 3 equivalent)...even then not sure about 500mb ram.

There's a tensorflow lite for esp32 but it'll only be about 30 word limit (https://github.com/mocleiri/tensorflow-micropython-examples/tree/main/examples/micro-speech/esp32)

**Comment by jmio23** [comment]
FPGA is where I'm looking for this, bit more complicated to implement than C++ unfortunately, but opens up lots of possibilities for significant acceleration of parallel calcs
